tgui/packages/tgui/interfaces/TurretControl.js

```diff
@@ -0,0 +1,108 @@
+import { Window } from '../layouts';
+import { useBackend, useLocalState } from '../backend';
+import { Button, Section, Flex, Box, Stack } from '../components';
+
+const randInt = (a, b) => {
+  if (a > b) {
+    let temp = a;
+    a = b;
+    b = temp;
+  }
+  return Math.floor(Math.random() * (b - a + 1)) + a;
+};
+
+// this is a totally real and normal semicolon I promise
+const glitches = ['$', '{', ']', '%', '^', '?', '>', '¬', 'π', ';', 'и', 'ю', '/', '#', '~'];
+const glitch = (text, amount) => {
+  for (let i = 0; i < amount; i++) {
+    let index = randInt(0, text.length);
+    text = text.slice(0, index) + glitches[randInt(0, glitches.length - 1)] + text.slice(index, text.length);
+  }
+  return text;
+};
```

Comment:
```suggestion
const glitch = (text, amount) => {
  const chars = text.split('');
  for (let i = 0; i < amount; i++) {
    const charIndex = randInt(0, chars.length);
    chars[charIndex] = glitches[randInt(0, glitches.length - 1)];
  }
  return chars.join('');
};
```
![image](https://user-images.githubusercontent.com/62817778/167282066-1be60f2a-71ee-4a2a-a888-02c062a38357.png)

Repeated slicing to update characters felt wrong for the level of character update you were doing (i.e. length of original string vs. amount of characters corrupted). Yours is definitely better for less corrupted strings, but as you have a single-use here we can optimize for it a bit I guess?